code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -339,252 +387,300 @@
 	R.add_language("Clownish",1)
 	R.add_language("Neo-Russkiya", 1)
 
-
+// Mining cyborg module.
 /obj/item/robot_module/miner
 	name = "miner robot module"
 	module_type = "Miner"
-	module_actions = list(
-		/datum/action/innate/robot_sight/meson,
-	)
+	module_actions = list(/datum/action/innate/robot_sight/meson)
 	custom_removals = list("KA modkits")
+	basic_modules = list(
+		/obj/item/storage/bag/ore/cyborg,
+		/obj/item/pickaxe/drill/cyborg,
+		/obj/item/shovel,
+		/obj/item/weldingtool/mini,
+		/obj/item/extinguisher/mini,
+		/obj/item/storage/bag/sheetsnatcher/borg,
+		/obj/item/t_scanner/adv_mining_scanner/cyborg,
+		/obj/item/gun/energy/kinetic_accelerator/cyborg,
+		/obj/item/gps/cyborg
+	)
+	emag_modules = list(/obj/item/borg/stun, /obj/item/pickaxe/drill/cyborg/diamond)
+	special_rechargables = list(/obj/item/extinguisher/mini, /obj/item/weldingtool/mini)
 
-/obj/item/robot_module/miner/New()
-	..()
-	modules += new /obj/item/storage/bag/ore/cyborg(src)
-	modules += new /obj/item/pickaxe/drill/cyborg(src)
-	modules += new /obj/item/shovel(src)
-	modules += new /obj/item/weldingtool/mini(src)
-	modules += new /obj/item/extinguisher/mini(src)
-	modules += new /obj/item/storage/bag/sheetsnatcher/borg(src)
-	modules += new /obj/item/t_scanner/adv_mining_scanner/cyborg(src)
-	modules += new /obj/item/gun/energy/kinetic_accelerator/cyborg(src)
-	modules += new /obj/item/gps/cyborg(src)
-	emag = new /obj/item/borg/stun(src)
-
-	fix_modules()
+// Replace their normal drills with diamond drills
+/obj/item/robot_module/miner/emag_act()
+	var/obj/item/pickaxe/drill/cyborg/D = locate() in modules
+	basic_modules -= D // Remove this here so it won't get re-created in the module rebuild.
```

Comment:
Won't this break the borg if he gets unemagged?

Reply:
Yeah this will break now that I think about it. The miner borg is the snowflake of the group, its the only one that gets an item removed when emagged. Would simplify things if I could just add the diamond one and have two drills.